app/controllers/notifications_controller.rb

```diff
@@ -4,20 +4,24 @@ class NotificationsController < ApplicationController
   before_action :render_home_page_unless_authenticated, only: [:index]
 
   def index
-    scope = current_user.notifications
-    scope = params[:archive].present? ? scope.archived : scope.inbox
+    scope    = current_user.notifications
+    @starred = scope.starred.count
+
+    scope = if params[:starred].present?
```

Comment:
I think this may be slightly more readable as an if, else if instead of an if with an embedded ternary.

```
if params[:starred].present?
	scope = scope.starred
elsif params[:archive].present?
	scope =  scope.archived
else
	scope = scope.inbox
end
```

The obvious downside is the repetition of `scope=`